### PR TITLE
Replace Jaeger tracing with OpenTelemetry and make Jaeger's backend

### LINKED ISSUE #40456: deploy-sourcegraph (k8s): make Jaeger optional, always deploy otel-collector

### PR DESCRIPTION
**Pull request by sanderginn**
This PR replaces Jaeger's agent sidecars and collector with OpenTelemetry Collector, as a [combined DaemonSet and Deployment](https://opentelemetry.io/docs/collector/deployment/) configuration. It disables Jaeger's backend by default but offers an opt-in (through an overlay), which will also configure the OT Collector to export to Jaeger.

PR for documentation: https://github.com/sourcegraph/sourcegraph/pull/40936

Closes https://github.com/sourcegraph/sourcegraph/issues/40456

### Todo
- [x] Make Jaeger opt-in
- [x] Document configuration of OT Collector 

### Checklist

* [x] [CHANGELOG.md](https://github.com/sourcegraph/sourcegraph/blob/main/CHANGELOG.md) updated: [link](https://github.com/sourcegraph/sourcegraph/pull/40936/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed)
* [x] [K8s Upgrade notes updated](https://github.com/sourcegraph/sourcegraph/blob/main/doc/admin/updates/kubernetes.md): [link](https://github.com/sourcegraph/sourcegraph/pull/40936/files#diff-750037a0097700b48b97d291619cddf329ca663c1025675596f5aa536fc12e5b)
* [x] Sister [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker) change:
    - https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/848
    - https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/849 
* [x] All images have a valid tag and SHA256 sum

### Test plan

e2e testing with @bobheadxi on a fully operational cluster

### CODE REVIEW

File: configure/otel-collector/otel-collector.ConfigMap.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    deploy: sourcegraph
+    app.kubernetes.io/component: otel-collector
+    sourcegraph-resource-requires: no-cluster-admin
+data:
+  config.yaml: |
+    receivers:
+      otlp:
+        protocols:
+          grpc: # port 4317
+          http: # port 4318
```
Comment (bobheadxi) [inline configure/otel-collector/otel-collector.ConfigMap.yaml:15]: Should we note that these (and certain other things) should not be changed?

Comment (jhchabran) [reply configure/otel-collector/otel-collector.ConfigMap.yaml:15]: Agreed. That being said, how is this working? I see a comment character here 🤔

Comment (sanderginn) [reply configure/otel-collector/otel-collector.ConfigMap.yaml]: Fixed. I explained this to you on our call, but for visibility: empty values here default to those ports :) clarified it a bit

**Review by bobheadxi — APPROVED** [review]
LGTM on paper, let's run this through tomorrow!

**Pull request merged**